pruner/state.go

```diff
@@ -0,0 +1,81 @@
+package pruner
+
+import (
+	"fmt"
+	"sync"
+
+	quamina "quamina/lib"
+)
+
+// State represents the required capabilities for maintaining the set
```

Comment:
Insert grumble about short name here…

Reply:
Even I'll admit that this name is a little vague. Hmm. `LiveSet`, `LivePatterns`, `LiveSetState`. `PrunerState` is both redundant (according to me) and wrong (since a Pruner has more state that that). Maybe `PatternState`. Yah, I know it seems like I'd argue against myself there. Or is it `PatternsState` anyway?